manifest/schema2/manifest.go

```diff
@@ -20,6 +20,10 @@ const (
 	// MediaTypeLayer is the mediaType used for layers referenced by the
 	// manifest.
 	MediaTypeLayer = "application/vnd.docker.image.rootfs.diff.tar.gzip"
+
+	// MediaTypeForeignLayer is the mediaType used for layers that must be
+	// downloaded from foreign URLs.
+	MediaTypeForeignLayer = "application/vnd.docker.image.foreign.rootfs.diff.tar.gzip"
```

Comment:
@stevvooe @aaronlehmann If we make `URLs` part of every `Descriptor`, does it still make sense to have a separate media type for these "foreign" layers?